File: jsonToProperty_oc/ViewController.m
//
//  ViewController.m
//  jsonToProperty_oc
//
//  Created by CityMedia on 2018/10/31.
//  Copyright © 2018年 free. All rights reserved.
//

#import "ViewController.h"
@interface ViewController()

@property (unsafe_unretained) IBOutlet NSTextView *textView;
@property (unsafe_unretained) IBOutlet NSTextView *textViewR;
@property (unsafe_unretained) IBOutlet NSTextView *textViewEditL;
@property (unsafe_unretained) IBOutlet NSTextView *textViewEditR;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
//    a

    // Do any additional setup after loading the view.
}
- (IBAction)btnAction:(id)sender {
    
    NSString * stra = self.textView.textStorage.string;
    
    stra = [stra stringByReplacingOccurrencesOfString:@" " withString:@""];
    stra = [stra stringByReplacingOccurrencesOfString:@"\n" withString:@""];
    stra = [stra stringByReplacingOccurrencesOfString:@"\t" withString:@""];
    stra = [stra stringByReplacingOccurrencesOfString:@"{" withString:@""];
    stra = [stra stringByReplacingOccurrencesOfString:@"}" withString:@""];
    
    NSString * sepStr = @";";
    
    if (self.textViewEditL.textStorage.string.length > 0) {
        sepStr = self.textViewEditL.textStorage.string;
    }
    NSArray * arr = [stra componentsSeparatedByString:sepStr];
    
    NSString * strDef = @"";
    for (NSString * str in arr) {
        if (str.length > 0 && ![str isEqualToString:@" "] ) {
        }else{
            break;
        }
        
        
        NSString * setStra = @"=";
        if (self.textViewEditR.textStorage.string.length > 0) {
            setStra = self.textViewEditR.textStorage.string;
        }
        NSArray * arrKeyValue =  [str componentsSeparatedByString:setStra];
        
//        NSArray * arrKeyValue = [str componentsSeparatedByString:@"="];
        NSString * key = arrKeyValue[0];
        NSString * value = [arrKeyValue lastObject] ;
        
        if ([value rangeOfString:@"\\U"].location != NSNotFound) {
            value = [self stringByReplaceUnicode:value];
        }
        NSString * lastStr = @"/**";
        lastStr = [lastStr stringByAppendingString:@"\n"];
        lastStr = [lastStr stringByAppendingString:@" * "];
        lastStr = [lastStr stringByAppendingString:key];
        lastStr = [lastStr stringByAppendingString:@"  "];
        lastStr = [lastStr stringByAppendingString:value];
        lastStr = [lastStr stringByAppendingString:@"\n"];
        lastStr = [lastStr stringByAppendingString:@"*/"];
        lastStr = [lastStr stringByAppendingString:@"\n"];
        lastStr = [lastStr stringByAppendingString:@"@property(nonatomic,copy) NSString * "];
        lastStr = [lastStr stringByAppendingString:key];
        lastStr = [lastStr stringByAppendingString:@";"];
        lastStr = [lastStr stringByAppendingString:@"\n"];
        strDef = [strDef stringByAppendingString:lastStr];
    }
    self.textViewR.string = strDef;
    
    
}


- (NSString *)stringByReplaceUnicode:(NSString *)unicodeString
{
    NSMutableString *convertedString = [unicodeString mutableCopy];
    [convertedString replaceOccurrencesOfString:@"\\U" withString:@"\\u" options:0 range:NSMakeRange(0, convertedString.length)];
    CFStringRef transform = CFSTR("Any-Hex/Java");
    CFStringTransform((__bridge CFMutableStringRef)convertedString, NULL, transform, YES);
    
    return convertedString;
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

/**
 {
 academyId = 1;
 academyName = "\U6570\U7406\U5b66\U9662";
 }
 */




@end

File: swift/JsonToProperty_mac/JsonToProperty/ViewController.swift
//
//  ViewController.swift
//  JsonToProperty
//
//  Created by CityMedia on 2018/10/18.
//  Copyright © 2018年 free. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
    }
  
    
    @IBOutlet var textView: NSTextView!
    @IBOutlet var textViewR: NSTextView!
    
    @IBAction func btnAction(_ sender: NSButton) {
        
        var str : String = textView.textStorage?.string ?? ""
        
        str = str.replacingOccurrences(of: " ", with: "")
        str = str.replacingOccurrences(of: "\n", with: "")
        str = str.replacingOccurrences(of: "{", with: "")
        str = str.replacingOccurrences(of: "}", with: "")
        let arr:[String] = str.components(separatedBy: ";")
        
        
        var strDef : String = "";
        for substr:String in arr {
            if substr.isEmpty{
                break
            }
            
            let arr = substr.components(separatedBy: "=")
            let key = arr[0]
            let value = arr.last
            
            var lastStr = "/**";
            lastStr = lastStr.appending("\n")
            lastStr = lastStr.appending(" * ")
            lastStr = lastStr.appending(key)
            lastStr = lastStr.appending("  ")
            lastStr = lastStr.appending(value ?? "value")
            lastStr = lastStr.appending("\n")
            lastStr = lastStr.appending("*/")
            lastStr = lastStr.appending("@property(nonatomic,copy) NSString * ")
            lastStr = lastStr.appending(key)
            lastStr = lastStr.appending(";")
            lastStr = lastStr.appending("\n")
            strDef = strDef.appending(lastStr)
        }
        
        
        textViewR.string = strDef
        
        
        print("btn click")
        
    }
    
    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    


}

/**
 {
 academyId = 1;
 academyName = "\U6570\U7406\U5b66\U9662";
 average = 0;
 cloStartNumber = 1;
 cloStopNumber = 1;
 cloTotalNumber = 1;
 collegeId = 1;
 courseCate = "\U516c\U5171\U5fc5\U4fee\U8bfe";
 courseCode = 201901;
 courseName = testCLO;
 courseScore = 4;
 courseStatus = "<null>";
 courseTimes = "<null>";
 courseType = "\U7406\U8bba\U6559\U5b66";
 createTime = "2018-10-23 10:26:02";
 createUser = 22;
 deleteTime = 0;
 id = 46;
 isRelease = 1;
 majorId = 1;
 majorName = "\U4fe1\U606f\U4e0e\U8ba1\U7b97\U79d1\U5b66";
 releaseTime = "2018-10-30 14:24:39";
 remark = test;
 semester = 1;
 status = 1;
 updateTime = "2018-10-30 14:24:39";
 updateUser = 1;
 year = 2019;
 }
 */

